### PR TITLE
Add copernicus dataspace watcher

### PR DESCRIPTION
**Pull request by mraspaud**
This PR adds a copernicus dataspace watcher through polling.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,6 +21,7 @@ minio = ["minio", "s3fs"]
 local = ["watchdog"]
 publishing = ["posttroll"]
 ssh = ["paramiko"]
+dataspace = ["oauthlib", "requests_oauthlib","s3fs"]
```
Comment (pnuu) [inline pyproject.toml:24]:
```suggestion
dataspace = ["oauthlib", "requests_oauthlib", "s3fs"]
```

File: src/pytroll_watchers/dataspace_watcher.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    last_pub_date = datetime.datetime.now(datetime.timezone.utc) - start_from
+
+    for next_check in run_every(polling_interval):
+        generator = generate_download_links_since(filter_string, dataspace_auth, last_pub_date, storage_options)
+        for s3path, metadata in generator:
+            last_pub_date = update_last_publication_date(last_pub_date, metadata)
+            yield s3path, metadata
+        logger.info("Finished polling.")
+        if next_check > datetime.datetime.now(datetime.timezone.utc):
+            logger.info(f"next iteration at {next_check}")
```
Comment (pnuu) [inline src/pytroll_watchers/dataspace_watcher.py:84]:
```suggestion
            logger.info(f"Next iteration at {next_check}")
```

File: src/pytroll_watchers/dataspace_watcher.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        last_publication_date = publication_date
+    return last_publication_date
+
+
+def _fromisoformat(metadata):
+    try:
+        return datetime.datetime.fromisoformat(metadata["PublicationDate"])
+    except ValueError:
+        # for python 3.9
+        return datetime.datetime.strptime(metadata["PublicationDate"], "%Y-%m-%dT%H:%M:%S.%f%z")
```
Comment (pnuu) [inline src/pytroll_watchers/dataspace_watcher.py:119]: The try/execpt is not needed, pyproject.toml defines >= 3.10.

```suggestion
    return datetime.datetime.fromisoformat(metadata["PublicationDate"])
```

**Review by pnuu — COMMENTED** [review]
Some suggestions.

File: src/pytroll_watchers/dataspace_watcher.py (hunk 3 above)
Comment (mraspaud) [reply src/pytroll_watchers/dataspace_watcher.py:119]: Then I need to fix the ci.yaml file, thanks for spotting this

Comment (mraspaud) [reply src/pytroll_watchers/dataspace_watcher.py:119]: Oh, my bad, the incompatibility is actually with 3.10 also, not just 3.9
```suggestion
    try:
        return datetime.datetime.fromisoformat(metadata["PublicationDate"])
    except ValueError:
        # for python 3.10
        return datetime.datetime.strptime(metadata["PublicationDate"], "%Y-%m-%dT%H:%M:%S.%f%z")
```

**Pull request merged**